src/Faker/Provider/DateTime.php

```diff
@@ -100,29 +113,67 @@ public static function time($format = 'H:i:s', $max = 'now')
      *
      * @param string $startDate Defaults to 30 years ago
      * @param string $endDate   Defaults to "now"
+     *
      * @example DateTime('1999-02-02 11:42:52')
+     *
      * @return \DateTime
      */
     public static function dateTimeBetween($startDate = '-30 years', $endDate = 'now')
     {
-        $startTimestamp = $startDate instanceof \DateTime ? $startDate->getTimestamp() : strtotime($startDate);
-        $endTimestamp = static::getMaxTimestamp($endDate);
+        if (!$startDate instanceof \DateTime) {
+            $startDate = new \DateTime($startDate);
+        }
+        if (!$endDate instanceof \DateTime) {
+            $endDate = new \DateTime($endDate);
+        }
+        $returnDate = $startDate;
```

Comment:
You need to `clone()`, otherwise you'll be altering the `$startDate` parameter, which is unexpected.